### PR TITLE
[Filesystem] Reduce complexity of ->remove()

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q | A |
| --- | --- |
| Branch | 2.3 |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | no |
| Fixed tickets | - |
| License | MIT |
| Doc PR | - |

### CODE REVIEW

File: src/Symfony/Component/Filesystem/Filesystem.php [hunk 1]
```diff
@@ -143,34 +143,22 @@ public function remove($files)
...
                 }
-            } else {
-                // https://bugs.php.net/bug.php?id=52176
-                if ('\\' === DIRECTORY_SEPARATOR && is_dir($file)) {
-                    if (true !== @rmdir($file)) {
-                        throw new IOException(sprintf('Failed to remove file %s', $file));
-                    }
-                } else {
-                    if (true !== @unlink($file)) {
-                        // handle broken symlinks on Windows systems
```
Comment (stof) [inline src/Symfony/Component/Filesystem/Filesystem.php]: is this case still handled by the new code ?

File: src/Symfony/Component/Filesystem/Filesystem.php [hunk 2]
```diff
@@ -143,34 +143,22 @@ public function remove($files)
         $files = iterator_to_array($this->toIterator($files));
         $files = array_reverse($files);
         foreach ($files as $file) {
-            if (!$this->exists($file) && !is_link($file)) {
-                continue;
-            }
-
-            if (is_dir($file) && !is_link($file)) {
+            if (is_link($file)) {
+                if (!@unlink($file) && !@rmdir($file)) {
```
Comment (stof) [inline src/Symfony/Component/Filesystem/Filesystem.php]: ``rmdir`` is only for https://bugs.php.net/bug.php?id=52176, right ?

File: src/Symfony/Component/Filesystem/Filesystem.php (hunk 1 above)
Comment (nicolas-grekas) [inline src/Symfony/Component/Filesystem/Filesystem.php]: yes, in the `is_link` case, where unlink and rmdir dir are both tried

File: src/Symfony/Component/Filesystem/Filesystem.php (hunk 2 above)
Comment (nicolas-grekas) [inline src/Symfony/Component/Filesystem/Filesystem.php]: yes

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

**Pull request merged**